dialogflow/session_entity_type_management.rb

```diff
@@ -0,0 +1,126 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require 'securerandom'
+
+def list_session_entity_types project_id:, session_id:
+  # [START dialogflow_list_session_entity_types]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  
+  require "google/cloud/dialogflow"
+
+  session_entity_types_client = Google::Cloud::Dialogflow::SessionEntityTypes.new
+  session_path = session_entity_types_client.class.session_path project_id,
+                                                                session_id
+
+  session_entity_types = session_entity_types_client.list_session_entity_types session_path
+
+  puts "SessionEntityTypes for session #{session_path}:\n"
+  session_entity_types.each do |session_entity_type|
+    entity_values = session_entity_type.entities.map {|e| e.value}
+    puts "=" * 20
+    puts "SessionEntityType name: #{session_entity_type.name}"
+    puts "Numer of entities:      #{session_entity_type.entities.size}"
+    puts "SessionEntityType entity values: #{entity_values}"
+  end
+  # [END dialogflow_list_session_entity_types]
+end
+
+
+def create_session_entity_type project_id:, session_id:,
+                               entity_type_display_name:,
+                               entity_values:
+  # [START dialogflow_create_session_entity_type]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  # entity_type_display_name = "existing_session_entity_type_display_name"
```

Comment:
what is the display name value for?

Reply:
it is used to identify an existing entity type to be overridden.  A "session entity type" really is a temporary modification of an existing entity type.

unfortunately this seems the only place where an entity type is identified with its (unique) display name instead of its id.